mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -0,0 +1,479 @@
+        # print Wi
+        result = result + 'W' + ','
+        # print K
+        result = result + 'K' + ','
+        # print Y
+        result = result + 'Y' + ','
+        # print X
+        result = result + 'X' + ','
+        # print dilation height
+        result = result + 'DilationH' + ','
+        # print dilation width
+        result = result + 'DilationW' + ','
+        # print stride height
+        result = result + 'StrideH' + ','
+        # print stride width
+        result = result + 'StrideW' + ','
+        # print padding width
+        result = result + 'PaddingH' + ','
+        # print padding height
+        result = result + 'PaddingW' + ','
+
+        # benchmarking fields
+
+        # print TFlops
+        result = result + 'TFlops'
+        return result
+
+    def generateCSVContent(self, nanoSeconds):
+        result = ''
+        # print operation.
+        result = result + self.direction + ','
+        # print data type.
+        result = result + self.dataType + ','
+        # print XDLOPS
+        result = result + str(self.xdlops) + ','
+        # print filter layout.
+        result = result + self.filterLayout + ','
+        # print input layout.
+        result = result + self.inputLayout + ','
+        # print output layout.
+        result = result + self.outputLayout + ','
+        # print N
+        result = result + str(self.n) + ','
+        # print C
+        result = result + str(self.c) + ','
+        # print Hi
+        result = result + str(self.hi) + ','
+        # print Wi
+        result = result + str(self.wi) + ','
+        # print K
+        result = result + str(self.k) + ','
+        # print Y
+        result = result + str(self.y) + ','
+        # print X
+        result = result + str(self.x) + ','
+        # print dilation height
+        result = result + str(self.dilationH) + ','
+        # print dilation width
+        result = result + str(self.dilationW) + ','
+        # print stride height
+        result = result + str(self.convStrideH) + ','
+        # print stride width
+        result = result + str(self.convStrideW) + ','
+        # print padding width
+        result = result + str(self.paddingH) + ','
+        # print padding height
+        result = result + str(self.paddingW) + ','
+
+        # benchmarking fields
+
+        # print TFlops
+        result = result + str(self.computeTFlops(nanoSeconds))
+        return result
+
+    def generateMlirDriverCommandLine(self):
+        result = ''
+        # set operation.
+        if self.direction == 'fwd':
+            result = result + '--operation conv2d'
+        elif self.direction == 'bwd':
+            result = result + '--operation conv2d_bwd_data'
+        elif self.direction == 'wrw':
+            result = result + '--operation conv2d_bwd_weight'
+        # set data type.
+        result = result + ' -t ' + self.dataType
+        # set XDLOPS
+        if self.xdlops == True:
+            result = result + ' -x2'
```

Comment:
For these conditionals, when you're refactoring, you can use `if self.xdlops then ' -x2' else ''`